src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -280,7 +280,7 @@ protected override void Dispose (bool disposing)
 			}
 		}
 
-		bool allowsCellularAccess;
+		bool allowsCellularAccess = true;
```

Comment:
I think that honoring the NSUrlSessionConfiguration passed to the instance would be better.

In other words, add the following line here https://github.com/xamarin/xamarin-macios/blob/ef4f329fa813b09efabdd029213db571b1c02788/src/Foundation/NSUrlSessionHandler.cs#L151-L163:

```csharp
allowsCellularAccess = configuration.AllowsCellularAccess;
```

Reply:
agreed - the default `.ctor` calls this one (and that 2nd .ctor needs to consider it too)